wrangler-core/src/main/java/io/cdap/wrangler/executor/RecipePipelineExecutor.java

```diff
@@ -122,14 +136,25 @@ public List<Row> execute(List<Row> rows) throws RecipeException {
               if (cumulativeRows.size() < 1) {
                 break;
               }
+              if (schema != null) {
+                schema = directive.getOutputSchema(schema) != null ? directive.getOutputSchema(schema) :
+                  generateOutputSchema(schema, cumulativeRows);
+              }
             } catch (ReportErrorAndProceed e) {
               messages.add(String.format("%s (ecode: %d)", e.getMessage(), e.getCode()));
               collector
                 .add(new ErrorRecord(rows.subList(i, i + 1).get(0), String.join(",", messages), e.getCode(), true));
               cumulativeRows = new ArrayList<>();
               break;
+            } catch (RecordConvertorException e) {
+              throw new RecipeException("Problem generating schema for a field: " + e.getMessage(), e);
             }
           }
+          if (schema != null) {
+            schema = context.getTransientStore().get(TransientStoreKeys.OUTPUT_SCHEMA) == null ?
+              schema : getSchemaUnion(context.getTransientStore().get(TransientStoreKeys.OUTPUT_SCHEMA), schema);
```

Comment:
Since some directives can generate output rows that have different columns (ex parsers), we need to take a union of previous row schema and current one. Have modified the variable names and added comments to add clarity